contracts/src/bridge/FastBridgeSenderToEthereum.sol

```diff
@@ -0,0 +1,145 @@
+    uint256 public currentTicketID = 1; // Zero means not set, start at 1.
+    mapping(uint256 => Ticket) public tickets; // The tickets by ticketID.
+
+    // ************************************* //
+    // *              Events               * //
+    // ************************************* //
+
+    /**
+     * The bridgers need to watch for these events and
+     * relay the messageHash on the FastBridgeReceiverOnEthereum.
+     */
+    event OutgoingMessage(uint256 indexed ticketID, address indexed target, bytes32 indexed messageHash, bytes message);
+
+    // ************************************* //
+    // *        Function Modifiers         * //
+    // ************************************* //
+
+    modifier onlyByGovernor() {
+        require(governor == msg.sender, "Access not allowed: Governor only.");
+        _;
+    }
+
+    constructor(address _governor, IFastBridgeReceiver _fastBridgeReceiver) SafeBridgeSenderToEthereum() {
+        governor = _governor;
+        fastBridgeReceiver = _fastBridgeReceiver;
+    }
+
+    // ************************************* //
+    // *         State Modifiers           * //
+    // ************************************* //
+
+    /**
+     * Sends an arbitrary message from one domain to another
+     * via the fast bridge mechanism
+     *
+     * @param _receiver The L1 contract address who will receive the calldata
+     * @param _calldata The receiving domain encoded message data.
+     * @return ticketID The identifier to provide to sendSafeFallback()
+     */
+    function sendFast(address _receiver, bytes memory _calldata) external override returns (uint256 ticketID) {
+        require(msg.sender == fastSender, "Access not allowed: Fast Sender only.");
+
+        ticketID = currentTicketID++;
+
+        (bytes32 messageHash, bytes memory messageData) = _encode(ticketID, _receiver, _calldata);
+        emit OutgoingMessage(ticketID, _receiver, messageHash, messageData);
+
+        tickets[ticketID] = Ticket({messageHash: messageHash, sentSafe: false});
+    }
+
+    /**
+     * Sends an arbitrary message from one domain to another
+     * via the safe bridge mechanism, which relies on the chain's native bridge.
+     *
+     * It is unnecessary during normal operations but essential only in case of challenge.
+     *
+     * It may require some ETH (or whichever native token) to pay for the bridging cost,
+     * depending on the underlying safe bridge.
+     *
+     * @param _ticketID The ticketID as provided by `sendFast()` if any.
+     * @param _receiver The L1 contract address who will receive the calldata
+     * @param _calldata The receiving domain encoded message data.
+     */
+    function sendSafeFallback(
+        uint256 _ticketID,
+        address _receiver,
+        bytes memory _calldata
+    ) external payable override {
+        Ticket storage ticket = tickets[_ticketID];
+        require(ticket.messageHash != 0, "Ticket does not exist.");
```

Comment:
Alternatively a null pattern can be used in the fastBridgeSender so that honest challengers can prove ticketIDs do not exist. Otherwise, challenges to claims on the fastBridgeReceiver about ticketIDs that do not exist cannot be resolved.

Moreover there is a *zero cost grief where the challenger and claimant are the same. A dishonest claimant can make a claim about a ticketID that does not exist, and the same dishonest actor can immediately challenge the claim. When the claim is eventually resolved, the challenger who is the same as the claimant can claim their own deposit back. Maybe 50% of the deposit should be burned or set towards rewarding future relayers / bridgers to ensure an immediate dishonest claim and challenge does not result in a zero cost delay grief.